02_knight_and_princess/kirill_tolstobrov/main.cpp

```diff
@@ -0,0 +1,165 @@
+    connections_map[id2][edges_amount - 1] = id1;
+  }
+
+  operator std::string() {
+    std::string result = "{\n  \"vertices\": [\n";
+
+    for (int i = 0; i < vertices_amount; i++) {
+      std::string vertex_string = "";
+      vertex_string += "    {\n      \"id\": " + std::to_string(i);
+      vertex_string += ",\n      \"edge_ids\": ";
+
+      auto& cur_vertex = connections_map[i];
+      for (auto it = cur_vertex.begin(); it != cur_vertex.end(); it++) {
+        if (it == cur_vertex.begin()) {
+          vertex_string += "[";
+        } else {
+          vertex_string += ", ";
+        }
+        vertex_string += std::to_string(it->first);
+      }
+      vertex_string += "]\n    }";
+
+      result += vertex_string;
+      if (i != vertices_amount - 1)
+        result += ",\n";
+    }
+
+    result += "\n  ],\n  \"edges\": [\n";
+
+    for (int i = 0; i < edges_amount; i++) {
+      std::string edge_string = "";
+      edge_string += "    {\n      \"id\": " + std::to_string(i);
+      edge_string += ",\n      \"vertex_ids\": ";
+
+      edge_string += "[" + std::to_string(edges[i].get_v1_id());
+      edge_string += ", " + std::to_string(edges[i].get_v2_id());
+      edge_string += "]\n    }";
+
+      result += edge_string;
+      if (i != edges_amount - 1)
+        result += ",\n";
+    }
+
+    result += "\n  ]\n}";
+    return result;
+  }
+};
+
+void generateGraph(Graph& g, int max_vertices = 10, int max_edges = 15) {
+  enum { PROBABILITY_COEF = 100 };
+
+  srand(std::time(NULL));
+
+  for (int i = 0; i < max_vertices; i++) {
+    const float new_vertex_prob =
+        PROBABILITY_COEF -
+        (float)PROBABILITY_COEF * g.get_vertices_amount() / max_vertices;
+    if (((double)std::rand() / RAND_MAX) * PROBABILITY_COEF < new_vertex_prob) {
+      g.add_new_vertex();
+      if (g.get_vertices_amount() == 1) {
+        continue;
+      }
+
+      int vertex_to_bind = std::rand() % (g.get_vertices_amount() - 1);
+      g.bind_vertices(g.get_vertices_amount() - 1, vertex_to_bind);
+    }
+  }
+
+  for (int i = g.get_edges_amount(); i < max_edges; i++) {
+    float new_edge_prob = PROBABILITY_COEF - (float)PROBABILITY_COEF *
+                                                 g.get_edges_amount() /
+                                                 max_edges;
+    if (((double)std::rand() / RAND_MAX) * PROBABILITY_COEF < new_edge_prob) {
+      int vertex_to_bind1 = std::rand() % g.get_vertices_amount();
+      int vertex_to_bind2 = std::rand() % g.get_vertices_amount();
+      g.bind_vertices(vertex_to_bind1, vertex_to_bind2);
+    }
+  }
+}
+
+void generateGraph(Graph& g,
+                   int vert_number,
+                   std::vector<std::pair<int, int>> connections) {
+  for (int i = 0; i < vert_number; i++) {
+    g.add_new_vertex();
+  }
+  for (int i = 0; i < connections.size(); i++) {
+    g.bind_vertices(connections[i].first, connections[i].second);
+  }
+}
+
+int main() {
+  Graph g;
```

Comment:
Никогда без нужды не стоит оставлять неинициализированные переменные
```suggestion
auto g = Graph();
```